libbeat/filter/filter.go

```diff
@@ -0,0 +1,167 @@
+package filter
+
+import (
+	"strings"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+type FilterCondition struct {
+}
+
+type DropFieldsConfig struct {
+	Fields []string `yaml:"fields"`
+}
+
+type IncludeFieldsConfig struct {
+	Fields []string `yaml:"fields"`
+}
+
+type FilterConfig struct {
+	DropFields    *DropFieldsConfig    `yaml:"drop_fields"`
+	IncludeFields *IncludeFieldsConfig `yaml:"include_fields"`
+}
+
+type FilterRule interface {
+	Filter(event *common.MapStr) *common.MapStr
```

Comment:
Agree that should not pass pointer. I wanted to do that anyway. But do you refer that needs to return no pointer as well? It's important to return a pointer as Filter might change the pointer as well.